api/src/main/java/org/pmiops/workbench/google/CloudBillingClientImpl.java

```diff
@@ -23,13 +25,32 @@ public CloudBillingClientImpl(
     this.retryHandler = retryHandler;
   }
 
-  public BillingAccount getBillingAccount(String billingAccountName) throws IOException {
+  @Override
+  public ProjectBillingInfo pollUntilBillingAccountLinked(
+      String billingAccountName, String projectId) throws IOException, InterruptedException {
+    Duration pollInterval = Duration.ofSeconds(2);
+    for (Instant deadline = Instant.now().plusSeconds(60);
```

Comment:
In practice, you're not likely to ever hit this deadline as it's already our global request deadline, I would suggest setting it lower